pkg/cmd/roachprod-bench/cluster/execute.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cluster
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+// RemoteCommand is a command to be executed on a remote node. The Metadata field is used to
+// store additional information from the original caller.
+type RemoteCommand struct {
+	Args     []string
+	Metadata interface{}
+}
+
+// RemoteResponse is the response to a RemoteCommand.
+type RemoteResponse struct {
+	RemoteCommand
+	Stdout   string
+	Stderr   string
+	Err      error
+	Duration time.Duration
+}
+
+func roachprodRunWithOutput(
+	log *logger.Logger, clusterName string, cmdArray []string,
+) (string, string, error) {
+	bufOut, bufErr := new(bytes.Buffer), new(bytes.Buffer)
+	err := roachprod.Run(context.Background(), log, clusterName, "", "", false, bufOut, bufErr, cmdArray)
```

Comment:
`roachprod.RunWithDetails` does this for you - the stderr and stdout will be in the `RunResultDetails`